perception/autoware_tensorrt_bevdet/src/postprocess.cu

```diff
@@ -0,0 +1,212 @@
+  int label = cls_range;       // Initialize as the first class of the task
+  for (int i = 1; i < num_class_in_task; ++i) {
+    float cur_score = cls[idx + i * map_size];
+    if (cur_score > max_score) {
+      max_score = cur_score;
+      label = i + cls_range;
+    }
+  }
+
+  int coor_x = idx % output_h;  //
+  int coor_y = idx / output_w;  //
+
+  float conf = sigmoid_gpu(max_score);  // Calculate confidence level
+  if (conf > score_thresh) {
+    int cur_valid_box_id = atomicAdd(res_box_num, 1);
+    res_box[cur_valid_box_id * kBoxBlockSize + 0] =
+      (reg[idx + 0 * map_size] + coor_x) * x_step + x_start;
+    res_box[cur_valid_box_id * kBoxBlockSize + 1] =
+      (reg[idx + 1 * map_size] + coor_y) * y_step + y_start;
+    res_box[cur_valid_box_id * kBoxBlockSize + 2] = hei[idx];
+    res_box[cur_valid_box_id * kBoxBlockSize + 3] =
+      expf(dim[idx + 0 * map_size]) * rescale_factor[label];  // nms scale
+    res_box[cur_valid_box_id * kBoxBlockSize + 4] =
+      expf(dim[idx + 1 * map_size]) * rescale_factor[label];
+    res_box[cur_valid_box_id * kBoxBlockSize + 5] =
+      expf(dim[idx + 2 * map_size]) * rescale_factor[label];
+    res_box[cur_valid_box_id * kBoxBlockSize + 6] = atan2f(rot[idx], rot[idx + map_size]);
+    res_box[cur_valid_box_id * kBoxBlockSize + 7] = vel[idx];
+    res_box[cur_valid_box_id * kBoxBlockSize + 8] = vel[idx + map_size];
+
+    res_conf[cur_valid_box_id] = conf;
+    res_cls[cur_valid_box_id] = label;
+  }
+}
+
+PostprocessGPU::PostprocessGPU(
+  const int _class_num, const float _score_thresh, const float _nms_thresh,
+  const int _nms_pre_maxnum, const int _nms_post_maxnum, const int _down_sample,
+  const int _output_h, const int _output_w, const float _x_step, const float _y_step,
+  const float _x_start, const float _y_start, const std::vector<int> & _class_num_pre_task,
+  const std::vector<float> & _nms_rescale_factor)
+: class_num(_class_num),
+  score_thresh(_score_thresh),
+  nms_thresh(_nms_thresh),
+  nms_pre_maxnum(_nms_pre_maxnum),
+  nms_post_maxnum(_nms_post_maxnum),
+  down_sample(_down_sample),
+  output_h(_output_h),
+  output_w(_output_w),
+  x_step(_x_step),
+  y_step(_y_step),
+  x_start(_x_start),
+  y_start(_y_start),
+  map_size(output_h * output_w),
+  class_num_pre_task(_class_num_pre_task),
+  nms_rescale_factor(_nms_rescale_factor),
+  task_num(_class_num_pre_task.size())
+{
+  CHECK_CUDA(cudaMalloc((void **)&boxes_dev, kBoxBlockSize * map_size * sizeof(float)));
+  CHECK_CUDA(cudaMalloc((void **)&score_dev, map_size * sizeof(float)));
+  CHECK_CUDA(cudaMalloc((void **)&cls_dev, map_size * sizeof(int)));
+  CHECK_CUDA(cudaMalloc((void **)&sorted_indices_dev, map_size * sizeof(int)));
+  CHECK_CUDA(cudaMalloc((void **)&valid_box_num, sizeof(int)));
+  CHECK_CUDA(cudaMalloc((void **)&nms_rescale_factor_dev, class_num * sizeof(float)));
+
+  CHECK_CUDA(cudaMallocHost((void **)&boxes_host, kBoxBlockSize * map_size * sizeof(float)));
```

Comment:
Similar to line 100,  add explicit error checks after each `cudaMallocHost` call to ensure proper memory allocation on the host side.